File: problems/algorithms/string-rna-transcription/js/response/rna-transcription.js
function toRna(dna) {
  let complement = {
    G: 'C',
    C: 'G',
    T: 'A',
    A: 'U',
  };

  let valid = nucleotide => Object.keys(complement).includes(nucleotide);

  let complementing = nucleotide => complement[nucleotide];

  let replacing = nucleotide => {
    if (valid(nucleotide)) return complementing(nucleotide);
    else throw new Error('Invalid input');
  };

  return dna
    .split('')
    .map(replacing)
    .join('');
}

export { toRna };

File: problems/algorithms/string-rna-transcription/ruby/response/src/rna_transcription.rb
class DNA
  def DNA.to_rna(dna)
    @complement = {
      'G' => 'C',
      'C' => 'G',
      'T' => 'A',
      'A' => 'U'
    }

    rna = ''
    dna.split('').each { |value|
      rna = rna + @complement[value]
    }
    rna
  end
end
